src/DataStreams/ParallelParsingBlockInputStream.cpp

```diff
@@ -195,19 +198,23 @@ void ParallelParsingBlockInputStream::parserThreadFunction(ThreadGroupStatusPtr
     }
     catch (...)
     {
-        onBackgroundException();
+        onBackgroundException(unit.offset);
     }
 }
 
-void ParallelParsingBlockInputStream::onBackgroundException()
+void ParallelParsingBlockInputStream::onBackgroundException(size_t offset)
 {
-    tryLogCurrentException(__PRETTY_FUNCTION__);
-
     std::unique_lock<std::mutex> lock(mutex);
     if (!background_exception)
     {
         background_exception = std::current_exception();
+
+        if (ParsingException * e = exception_cast<ParsingException *>(background_exception))
```

Comment:
```suggestion
        if (ParsingException * e = exception_cast<ParsingException *>(background_exception)
            && e->getLineNumber() != -1)
```

I'd use `std::optional` here because we don't care about overhead, but special value is also OK.

Reply:
It has type `ssize_t` just for having special value.

Reply:
Yep. We have to handle the special value here (when for some reason the row number was not set by the parser), hence the above diff -- does it look correct?

Reply:
In case that parser didn't set the row into exception, we wouldn't add any offset. Row number will contain the special value -1 and we will handle this state in `displayText()` method of exception (will insert empty string instead of `{}`.)

I think this is Ok., because the initial goal of this PR was to not to tell the wrong information about line with error.

Reply:
But in the next line we are adding the processing unit offset to it. If it was unset (-1), it will become the last line of the previous unit.

Reply:
Your commit suggestion confused me a bit. I thought, that it was my code in comparison with an old version.